src/command/expire/expire.c

```diff
@@ -0,0 +1,680 @@
+    int int2 = atoi(strPtr(strLstGet(archiveSort2, 1)));
+
+    return (int2 - int1);
+}
+
+static StringList *
+sortArchiveId(StringList *sortString, SortOrder sortOrder)
+{
+    FUNCTION_LOG_BEGIN(logLevelDebug);
+        FUNCTION_LOG_PARAM(STRING_LIST, sortString);
+        FUNCTION_LOG_PARAM(ENUM, sortOrder);
+    FUNCTION_LOG_END();
+
+    ASSERT(sortString != NULL);
+
+    switch (sortOrder)
+    {
+        case sortOrderAsc:
+        {
+            lstSort((List *)sortString, archiveIdAscComparator);
+            break;
+        }
+
+        case sortOrderDesc:
+        {
+            lstSort((List *)sortString, archiveIdDescComparator);
+            break;
+        }
+    }
+
+    FUNCTION_LOG_RETURN(STRING_LIST, sortString);
+}
+
+typedef struct ArchiveRange
+{
+    const String *start;
+    const String *stop;
+} ArchiveRange;
+
+/***********************************************************************************************************************************
+Common function for expiring any backup
+***********************************************************************************************************************************/
+static void
+expireBackup(InfoBackup *infoBackup, String *removeBackupLabel, String *backupExpired)
+{
+    FUNCTION_LOG_BEGIN(logLevelDebug);
+        FUNCTION_LOG_PARAM(INFO_BACKUP, infoBackup);
+        FUNCTION_LOG_PARAM(STRING, removeBackupLabel);
+        FUNCTION_LOG_PARAM(STRING, backupExpired);
+    FUNCTION_LOG_END();
+
+    ASSERT(infoBackup != NULL);
+    ASSERT(removeBackupLabel != NULL);
+    ASSERT(backupExpired != NULL);
+
+    storageRemoveNP(
+        storageRepoWrite(),
+        strNewFmt(STORAGE_REPO_BACKUP "/%s/%s", strPtr(removeBackupLabel), INFO_MANIFEST_FILE));
+
+    storageRemoveNP(
+        storageRepoWrite(),
+        strNewFmt(STORAGE_REPO_BACKUP "/%s/%s", strPtr(removeBackupLabel), INFO_MANIFEST_FILE INFO_COPY_EXT));
+
+    // Remove the backup from the info file
+    infoBackupDataDelete(infoBackup, removeBackupLabel);
+
+    if (strSize(backupExpired) == 0)
+        strCat(backupExpired, strPtr(removeBackupLabel));
+    else
+        strCatFmt(backupExpired, ", %s", strPtr(removeBackupLabel));
+
+    FUNCTION_LOG_RETURN_VOID();
+}
+
+/***********************************************************************************************************************************
+Expire differential backups
+***********************************************************************************************************************************/
+static unsigned int
+expireDiffBackup(InfoBackup *infoBackup)
+{
+    FUNCTION_LOG_BEGIN(logLevelDebug);
+        FUNCTION_LOG_PARAM(INFO_BACKUP, infoBackup);
+    FUNCTION_LOG_END();
+
+    ASSERT(infoBackup != NULL);
+
+    unsigned int result = 0;
+
+    MEM_CONTEXT_TEMP_BEGIN()
+    {
+        // ??? Retention options will need to be indexed
+        unsigned int differentialRetention = cfgOptionTest(cfgOptRepoRetentionDiff) ?
```

Comment:
Why is ??? still here?

Reply:
Was just a note to remind us to do it. Removed.